tools/constants/constants.go

```diff
@@ -24,3 +24,20 @@ const (
 	// Components is the placeholder for the resource type of components
 	Components ResourceType = "Components"
 )
+
+// DefaultFloat32Value is the exported constant value for the default float32 value for an uninitialized float32
+// variable.
+const (
+	DefaultFloat32Value float32 = iota
+)
+
+const (
+	// VersionNotNeeded is a place holder to indicate that the schema version does not need to be specified.
+	VersionNotNeeded float32 = -18.0
+	// MinComponentYAMLVersion is the minimum schema version for the component
+	// YAML supported by this masonry toolchain.
+	MinComponentYAMLVersion float32 = 3.0
+	// MaxComponentYAMLVersion is the minimum schema version for the component
+	// YAML supported by this masonry toolchain.
+	MaxComponentYAMLVersion float32 = VersionNotNeeded
+)
```

Comment:
Maybe we should use major and minor versions as integers, and a string for the full version? My internal alarm went off about using floats where the comparison might be off by a small fraction, though it _shouldn't_ be a problem for us since we won't be doing math with them?